File: front/src/app/model/db/Contrib.ts
import { Attachement } from "./Attachement";
import { DbDate, DbId, DbInsert } from "./dbDefs";
import { Product } from "./Product";
import { Store } from "./Store";
import { User } from "./User";

export type ContribStatus = "ACTIVE" | "REVOKED" | "REMOVED"

export interface Contrib
{
    id: DbId

    // Grouped by
    product: DbInsert<Product>
    store: DbInsert<Store>

    // Detail view
    author: DbInsert<User>
    price: number
    date: DbDate

    // Optional detail view
    comment?: string
    attachements: DbInsert<Attachement>[]

    // View status
    status: ContribStatus
}

File: back/model/db/Contrib.go
package model


// due to no union/enums in language, ContribStatus = string
type ContribStatus string

type Contrib struct {
    id DbId

    // grouped by
    product DbRef[Product]
    store DbRef[Store]

    author DbRef[User]
    price float32
    date any // todo: define date

    // optional detail view
    comment *string
    attachments []DbRef[Attachment]

    // view status
    status ContribStatus

}

